flink-ml-lib/src/main/java/org/apache/flink/ml/classification/knn/KnnModel.java

```diff
@@ -0,0 +1,223 @@
+        return model.setModelData(modelDataTable);
+    }
+
+    /** This operator loads model data and predicts result. */
+    private static class PredictLabelFunction extends RichMapFunction<Row, Row> {
+        private final String featureCol;
+        private KnnModelData knnModelData;
+        private final int k;
+        private final String broadcastKey;
+
+        public PredictLabelFunction(String broadcastKey, int k, String featureCol) {
+            this.k = k;
+            this.broadcastKey = broadcastKey;
+            this.featureCol = featureCol;
+        }
+
+        @Override
+        public Row map(Row row) {
+            if (knnModelData == null) {
+                knnModelData =
+                        (KnnModelData)
+                                getRuntimeContext().getBroadcastVariable(broadcastKey).get(0);
+            }
+            DenseVector vector = (DenseVector) row.getField(featureCol);
+            if (vector == null) {
+                return Row.join(row, new Row(1));
+            }
+            Tuple2<List<Double>, List<Double>> tuple2 = findNeighbor(vector);
+            double percent = 1.0 / tuple2.f0.size();
+            Map<Double, Double> detail = new HashMap<>(0);
+            for (Double obj : tuple2.f0) {
+                detail.merge(obj, percent, Double::sum);
+            }
+            double max = 0.0;
+            double prediction = 0.0;
+            for (Map.Entry<Double, Double> entry : detail.entrySet()) {
+                if (entry.getValue() > max) {
+                    max = entry.getValue();
+                    prediction = entry.getKey();
+                }
+            }
+            return Row.join(row, Row.of(prediction));
+        }
+
+        /** Finds the nearest k neighbors from whole vectors in matrix. */
+        private Tuple2<List<Double>, List<Double>> findNeighbor(DenseVector input) {
+            PriorityQueue<Tuple2<Double, Double>> priorityQueue =
+                    new PriorityQueue<>(Comparator.comparingDouble(o -> -o.f0));
+            double d = 0.0;
+            for (int i = 0; i < input.size(); ++i) {
+                d += input.values[i] * input.values[i];
+            }
+            Tuple2<DenseVector, Double> sample = Tuple2.of(input, d);
+            DenseMatrix packedFeatures = knnModelData.packedFeatures;
+            double[] labelValues = knnModelData.labels.values;
+            double[] normValues = knnModelData.featureNorms.values;
+            DenseVector distanceVector = new DenseVector(packedFeatures.numCols());
+            BLAS.gemv(-2.0, packedFeatures, true, sample.f0, 0.0, distanceVector);
+            for (int i = 0; i < distanceVector.values.length; i++) {
+                distanceVector.values[i] =
+                        Math.sqrt(Math.abs(distanceVector.values[i] + sample.f1 + normValues[i]));
+            }
+            List<Tuple2<Double, Double>> distances = new ArrayList<>(labelValues.length);
```

Comment:
nits: could we give it a more meaningful name to improve readability, e.g. `distanceAndLabels`?